### PR TITLE
feat(replay): Adds click to search on component name

### PR DESCRIPTION
**Pull request by c298lee**
If the component that was clicked on is of type React Component Name, then you can click on it to search for replays that also have those clicks. 
We only have data on whether or not the clicked component has a component name, the parents in the tree come in as a string so there's no type data, so only the clicked component will have the ability to click to search. Since the entire HTML tree comes in as one string, some string manipulation was required.

image
image

Relates to: https://github.com/getsentry/sentry/issues/64673

### CODE REVIEW

File: static/app/components/replays/breadcrumbs/breadcrumbItem.tsx [hunk 1]
```diff
@@ -201,6 +202,43 @@ function CrumbErrorIssue({frame}: {frame: ErrorFrame}) {
...
+    isClickFrame(frame) && frame.data.node?.attributes['data-sentry-component'];
+  const lastComponentIndex =
+    description.lastIndexOf('>') === -1 ? 0 : description.lastIndexOf('>') + 2;
+
+  return (
+    <Description title={description} showOnlyOnOverflow isHoverable>
+      {componentName ? (
+        <Fragment>
+          <div style={{display: 'inline'}}>
+            {description?.toString().substring(0, lastComponentIndex)}
```
Comment (michellewzhang) [inline static/app/components/replays/breadcrumbs/breadcrumbItem.tsx:219]: do we need the `toString()` here if we pass it in as a string up above?

File: static/app/components/replays/breadcrumbs/breadcrumbItem.tsx [hunk 2]
```diff
@@ -201,6 +202,43 @@ function CrumbErrorIssue({frame}: {frame: ErrorFrame}) {
...
+  const location = useLocation();
+  const organization = useOrganization();
+
+  const componentName =
+    isClickFrame(frame) && frame.data.node?.attributes['data-sentry-component'];
+  const lastComponentIndex =
+    description.lastIndexOf('>') === -1 ? 0 : description.lastIndexOf('>') + 2;
+
+  return (
+    <Description title={description} showOnlyOnOverflow isHoverable>
```
Comment (michellewzhang) [inline static/app/components/replays/breadcrumbs/breadcrumbItem.tsx:215]: what happens if the there's a react component name & the description here overflows? will there be two layers of tooltips :o

**Review by michellewzhang — APPROVED** [review]

File: static/app/components/replays/breadcrumbs/breadcrumbItem.tsx (hunk 1 above)
Comment (c298lee) [reply static/app/components/replays/breadcrumbs/breadcrumbItem.tsx:219]: oh thanks for catching that!

File: static/app/components/replays/breadcrumbs/breadcrumbItem.tsx (hunk 2 above)
Comment (c298lee) [reply static/app/components/replays/breadcrumbs/breadcrumbItem.tsx]: I changed it so that there's 2 layers of tooltips :D

**Review by michellewzhang — APPROVED** [review]
thanks for making the changes!

File: static/app/components/replays/breadcrumbs/breadcrumbItem.tsx [hunk 3]
```diff
@@ -201,6 +213,43 @@ function CrumbErrorIssue({frame}: {frame: ErrorFrame}) {
...
+  const componentName =
+    isClickFrame(frame) && frame.data.node?.attributes['data-sentry-component'];
+  const lastComponentIndex =
+    description.lastIndexOf('>') === -1 ? 0 : description.lastIndexOf('>') + 2;
+
+  return (
+    <Fragment>
+      {componentName ? (
+        <Fragment>
+          <div style={{display: 'inline'}}>
```
Comment (billyvg) [inline static/app/components/replays/breadcrumbs/breadcrumbItem.tsx:229]: I think this can be a `span`, no need for inline styles

**Pull request merged**